shell/client/grainview.js

```diff
@@ -0,0 +1,482 @@
+var counter = 0;
+
+GrainView = function GrainView(grainId, path, query, hash, token, parentElement) {
+  // Owned grains:
+  // grainId, path, query, hash, dep.
+  //   callback sets error, openingSession on failure,
+  //                 grainId, sessionId, sessionSub on success.
+  //
+  // Sturdyref ApiTokens:
+  // grainId, path, query, hash, dep.
+  //   callback sets error, openingSession on failure
+  //                 grainId, sessionId, sessionSub on success.
+  //
+  // Token-only sessions:
+  // grainId, token, path, query, hash, dep
+  //   callback sets error, openingSession on failure
+  //                 grainId, sessionId, title, and session Sub on success
+
+  this._grainId = grainId;
+  this._originalPath = path;
+  this._originalQuery = query;
+  this._originalHash = hash;
+  this._token = token;
+
+  this._status = "closed";
+  this._revealIdentity = undefined; // set to true or false to make explicit
+  this._dep = new Tracker.Dependency();
```

Comment:
I guess we could make each of them a `ReactiveVar`, in hindsight.  It might also make reactive recomputation more granular.

On the other hand, in a couple places it's nice to be able to batch `changed()` calls, particularly when you want certain changes to be observed atomically.  I'd want to be sure this._status and this._error change "together" as viewed from outside, and probably the same for `_status`, `_sessionId`, and `_title`.

I'm not sure what guarantees Tracker gives about recomputation order - when you emit a `changed()`, does it let the current context finish running before running any of the rescheduled computations, or does it do work/recomputations as `changed()` is called?

I'm okay switching the mutable variables to ReactiveVar so long as you won't see inconsistent states externally - it'd certainly help keep us from missing `depend()` and `changed()` calls when we add code.